test/common/common-tty.js

```diff
@@ -0,0 +1,52 @@
+/* eslint-disable node-core/required-modules */
+'use strict';
+
+// Named this way for use with the required-modules eslint rule.
+
+// Utilities for TTY related tests.
+
+const assert = require('assert');
+
+// Check TTY status on module load.
+checkIsTTY();
```

Comment:
This conflicts with `getTTYfd` below, where we also allow all of stdio to be non-TTY if there are other ways to get a TTY.